common/src/main/java/com/sequenceiq/cloudbreak/ccm/endpoint/ServiceFamily.java

```diff
@@ -0,0 +1,60 @@
+package com.sequenceiq.cloudbreak.ccm.endpoint;
+
+import java.util.Optional;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Represents a family of services.
+ *
+ * @param <T> the type of service endpoint used by the family
+ */
+public interface ServiceFamily<T extends ServiceEndpoint> {
+
+    /**
+     * The service family for nginx web servers on gateways.
+     */
+    ServiceFamily<HttpsServiceEndpoint> GATEWAY =
```

Comment:
The GATEWAY and KNOX constants should not live in an interface, but instead in a constant utility class - maybe "ServiceFamilies".

Reply:
My opinion about modern Java interfaces is that they now allow for not introducing unnecessary utility classes and abstract classes, but I have decided to "go along to get along". :)